src/critical_section.rs

```diff
@@ -0,0 +1,29 @@
+#[cfg(all(cortex_m, feature = "critical-section-single-core"))]
+mod single_core_critical_section {
+    use critical_section::{set_impl, Impl, RawRestoreState};
+
+    use crate::interrupt;
+    use crate::register::primask;
+
+    struct SingleCoreCriticalSection;
+    set_impl!(SingleCoreCriticalSection);
+
+    unsafe impl Impl for SingleCoreCriticalSection {
+        unsafe fn acquire() -> RawRestoreState {
+            let active = primask::read().is_active();
+            if active {
```

Comment:
Do we need to condition disabling interrupts on them currently being active, or can we unconditionally disable them (while still returning whether they _were_ active for `release()`)?

Reply:
it's what the PR did, can't think of a reason either. Changed it to do it unconditionally.